### DISCUSSION TITLE
Detected Object Scoring Model

### DISCUSSION THREAD

**Discussion by kirsch33** [blakeblackshear/frigate#5175] [General]
Hi guys, was just reviewing the "reducing false positives" page about how Frigate basically takes the median score of a running 3 data point set. I am just curious regarding how this was arrived at instead of for example, discarding completely scores below the min_score (not setting to 0) and computing an average? 

If it was acceptable to have the detection running a few seconds behind the actual footage, computing the scores based on a larger running dataset (>~25) would allow you to do some more interesting things with the data (like reducing the min_score quite a bit, checking percent of data within plus/minus 1 std deviation of the mean)?

Just thinking out loud and curious on how the current system was decided on, and if theres any interest or plans to revisit it moving foward.

**Comment by NickM-27** [comment]
> If it was acceptable to have the detection running a few seconds behind the actual footage

Frigate highly prioritizes real time detection, for many use cases latent detection is much less useful

**Reply by kirsch33** [reply]
That's understandable, I've been tinkering with zones and object filter settings for a while trying to reduce false positives (and maintain detection of parked cars) and ran across this page. I get that my use case is not everyone elses, but for me 2-5s behind real-time footage would be worth it if it resulted in a more reliable accuracy of an object being a true positive.

I wasn't sure how in-depth other methods of computing a detected object were looked into.

**Reply by NickM-27** [reply]
> behind real-time footage would be worth it if it resulted in a more reliable accuracy of an object being a true positive.

That's what https://deploy-preview-4055--frigate-docs.netlify.app/integrations/plus is for

**Comment by blakeblackshear** [comment]
I arrived at the current approach by testing the current models against thousands of clips from historical false positives. Lots of spiders, moths, rain, snow, etc. I tried lots of different approaches. Generally speaking, lower scores are completely wrong so often that they just skew the results and frame by frame tracking. I am sure there are ways to improve it, but I would want to evaluate based on objective metrics on a large dataset like that.
